department-of-veterans-affairs/abd-vro
# GH Actions: restrict access to certain GH Actions

## What was the problem?

Anyone on the current team are allowed to do potentially dangerous actions, such as deploy to prod, via GH Action workflows. 

Associated tickets or Slack threads:
- #1961

## How does this fix it?

Only allow members of the [VRO-RESTRICTED](https://github.com/orgs/department-of-veterans-affairs/teams/vro-restricted/members) team to do potentially dangerous actions.

Affected workflows:
- `delete-old-releases.yml`
- `delete-published-images.yml`
- `delete-workflow-runs.yml`
- `deploy-secrets.yml` - restricted only when deploying to `prod`
- `update-deployment.yml` - restricted only when deploying to `prod`

## How to test this PR
If you are not in [VRO-RESTRICTED](https://github.com/orgs/department-of-veterans-affairs/teams/vro-restricted/members), then the above GH Action workflows should fail (when run on this PR's branch `yoom/restrict-access`) with a `Only VRO-RESTRICTED team members can run this action` message.
